env.sh

```diff
@@ -55,12 +55,19 @@ if [[ -z ${DORIS_GCC_HOME} ]]; then
 fi
 
 gcc_ver=`${DORIS_GCC_HOME}/bin/gcc -dumpfullversion -dumpversion`
-required_ver="5.3.1"
+required_ver="7.1.0"
 if [[ ! "$(printf '%s\n' "$required_ver" "$gcc_ver" | sort -V | head -n1)" = "$required_ver" ]]; then 
     echo "Error: GCC version (${gcc_ver}) must be greater than or equal to ${required_ver}"
     exit 1
 fi
 
+flex_ver=`flex --version | awk '{print $2}'`
```

Comment:
why do this check? and update flex version?

Reply:
Because the C++ code generated by flex uses the keyword ```register```.

From C++11 deprecated the ```register``` keyword.  

In C++17 forbidden use ```register``` keyword.  flex fix the problem after 2.6.0 version.

reference: https://github.com/westes/flex/issues/132

Reply:
Good explain!
Can you add this explain to the source code?

Reply:
Yes, of course.